lib/rex/socket/ssh_factory.rb

```diff
@@ -0,0 +1,46 @@
+module Rex
+  module Socket
+
+    # This class exists to abuse the Proxy capabilities in the Net::SSH library to allow the use of Rex::Sockets
+    # for the transport layer in Net::SSH. The SSHFactory object will respond to the #open method and create the
+    # {Rex::Socket::Tcp}
+    class SSHFactory
+
+      # @!attribute msfraemwork
+      #   @return [Object] The framework instance object
+      attr_accessor :msframework
```

Comment:
Everywhere else in the code, a framework object is just called `framework`. It seems weird to buck that convention.